### DISCUSSION TITLE
Is it possible to use minio s3-gateway without etcd

### DISCUSSION THREAD

**Discussion by zhijian-pro** [minio/minio#14489] [Q&A] [answered]
I noticed the latest code has this line comment `Enable IAM admin APIs if etcd is enabled, if not just enable basic operations such as profiling, server info etc.`
https://github.com/minio/minio/blob/master/cmd/gateway-main.go#:~:text=//%20Enable%20IAM%20admin,registerAdminRouter(router%2C%20false)
I don't understand why `IAMObjectStore` cannot be used to store iam information without etcd

**Comment by harshavardhana** [comment, marked as answer]
Sure it works but all the credentials are ephemeral @zhijian-pro
